overlord/ifacestate/handlers.go

```diff
@@ -227,6 +227,24 @@ func (m *InterfaceManager) doConnect(task *state.Task, _ *tomb.Tomb) error {
 		return err
 	}
 
+	// connect OS snap if snap name is omitted
+	if slotRef.Snap == "" {
+		slotRef.Snap = "ubuntu-core"
+	}
+
+	// connect the plug to slot with matching interface if slot is omitted
+	if slotRef.Name == "" {
+		plugIface := m.repo.Plug(plugRef.Snap, plugRef.Name)
```

Comment:
This feels like something we should resolve inside the repository.  The repository has a lock and can do this atomically.